File: lib-universalAdapter/src/main/java/com/llj/adapter/observable/SimpleListObserverListener.java
package com.llj.adapter.observable;

import androidx.annotation.Nullable;

/**
 * PROJECT:CommonAdapter
 * DESCRIBE: 用于PagerAdapterConverter刷新，统一用onGenericChange处理
 * Created by llj on 2017/2/11.
 */

public abstract class SimpleListObserverListener<T> implements ListObserverListener<T> {

    @Override
    public void onItemRangeChanged(ListObserver<T> observer, int startPosition, int itemCount,
        @Nullable Object payload) {
        onGenericChange(observer);
    }

    @Override
    public void onItemRangeChanged(ListObserver<T> observer, int startPosition, int itemCount) {
        onGenericChange(observer);
    }

    @Override
    public void onItemRangeInserted(ListObserver<T> observer, int startPosition, int itemCount) {
        onGenericChange(observer);
    }

    @Override
    public void onItemRangeRemoved(ListObserver<T> observer, int startPosition, int itemCount) {
        onGenericChange(observer);
    }
}

File: lib-universalAdapter/src/main/java/com/llj/adapter/observable/SimpleListObserverListener.kt
package com.llj.adapter.observable

/**
 * describe 用于PagerAdapterConverter刷新，统一用onGenericChange处理
 *
 * @author liulinjie
 * @date 2020/4/18 12:27 AM
 */
abstract class SimpleListObserverListener<T> : ListObserverListener<T> {
    override fun onItemRangeChanged(observer: ListObserver<T>, startPosition: Int, count: Int,
                                    payload: Any?) {
        onGenericChange(observer)
    }

    override fun onItemRangeChanged(observer: ListObserver<T>, startPosition: Int, count: Int) {
        onGenericChange(observer)
    }

    override fun onItemRangeInserted(observer: ListObserver<T>, startPosition: Int, count: Int) {
        onGenericChange(observer)
    }

    override fun onItemRangeRemoved(observer: ListObserver<T>, startPosition: Int, count: Int) {
        onGenericChange(observer)
    }
}